### ISSUE TITLE
Entrypoint rewriting logic is ignoring `cmd`

### ISSUE THREAD

**Issue by bobcatfish** [tektoncd/pipeline#574]
# Expected Behavior

I'm a bit confused about this to tell you the truth, but I think _think_ that when we are wrapping a container, we should be respecting _both_ the `entrypoint` and the `cmd`

# Actual Behavior

When we [retrieve an image's entrypoint](https://github.com/knative/build-pipeline/blob/f267ad53eb83b9e6f8e47a7c7066ec5f6ba0bdb5/pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go#L119) we do just that - we get the entrypoint only and ignore the `cmd`. So if an image is relying on the `cmd` to do the right thing, we wipe it out and completely ignore it.

# Steps to Reproduce the Problem

1. Create a Task with a step that tries to rely on an image's `cmd`, e.g. in this case the `cmd` for the `golang` image is `go`, so we shouldn't need to specify that explicitly,:

```yaml
  - name: install-ko
    image: golang
    args:
    - "get"
    - "github.com/google/go-containerregistry/cmd/ko"
```

2. Create a TaskRun that uses the above. It will fail with an error like:

```
could not start the process: exec: \"get\": executable file not found in $PATH
```
3. If you update the step to use `go` explicitly, it will work:

```yaml
  - name: install-ko
    image: golang
    command: ["go"]
    args:
    - "get"
    - "github.com/google/go-containerregistry/cmd/ko"
```

# Additional Info

* For reference: https://stackoverflow.com/questions/21553353/what-is-the-difference-between-cmd-and-entrypoint-in-a-dockerfile
* Note that in #564 @pivotal-nader-ziada is updating this logic

**Comment by pivotal-nader-ziada** [comment]
in pr #564, I have logic that will check the remote image `entrypoint` and if there isn't one, will check the `cmd` 
would that cover the problem you ran into?

**Comment by abayer** [comment]
Oh boy, this is always a fun area - we've flip-flopped on how the Jenkins/Docker integration works from using `entrypoint` vs `cmd`...well, multiple times. The lesson I've taken away from our pain there is that if you try to handle every particular permutation of `entrypoint` vs `cmd` as default behavior, you're going to have a bad time. My opinion is that it's best to settle on one expected/required behavior, and just accept that this means some images will require a bit more configuration to use. Our situation here seems to be much simpler than the Jenkins/Docker concerns, though, so I may just be paranoid. =)

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish I think this should be closed as part of the changes in PR #564

**Comment by dlorenc** [comment]
Fixed with #564
